Add support for specifying TLS protocols

ljacomet opened the pull request:
Following changes in Maven Central, only TLSv1.2 is accepted. While
there is a workaround for standard JDK SSL connections, the use of
HttpClient in Gradle did not allow any customization.
This will default TLS protocols to `["TLSv1", "TLSv1.1", "TLSv1.2"]` for Java 7.
It also adds support using the same system property as the one used by JDK
SSL.

It also adds that property to tests that do interact with Maven Central.

wolfs commented on subprojects/internal-performance-testing/src/integTest/groovy/org/gradle/performance/fixture/MavenDownloaderTest.groovy:
```diff
@@ -37,6 +38,9 @@ class MavenDownloaderTest extends Specification {
     def setup() {
         installRoot = tmpDir.newFolder()
         downloader = new MavenInstallationDownloader(installRoot)
+        if (JavaVersion.current().isJava7()) {
```
Is this still required?

ljacomet replied:
Yes, that test actually uses a standard JDK connection for the download.

wolfs commented on subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java:
```diff
@@ -102,7 +113,7 @@ public void configure(HttpClientBuilder builder) {
 
     private void configureSslSocketConnectionFactory(HttpClientBuilder builder, SslContextFactory sslContextFactory, HostnameVerifier hostnameVerifier) {
-        builder.setSSLSocketFactory(new SSLConnectionSocketFactory(sslContextFactory.createSslContext(), hostnameVerifier));
+        builder.setSSLSocketFactory(new SSLConnectionSocketFactory(sslContextFactory.createSslContext(), SSL_PROTOCOLS, null, hostnameVerifier));
```
Can we add a test to check that the right protocol version is added on Java 7?

ljacomet replied:
That makes sense, let me find a way

ljacomet commented:
@wolfs I have addressed your comments and also added support for `https.protocols` ... which lacks a test cause aside from defining a custom task, everything I tried inside the class is simply too late for the system property to be utilized.

ijuma commented:
Thanks for the fix. Any idea when 4.8.1 will be released?

marcphilipp commented:
It's planned for today but it might be tomorrow since we're running a bit late.
